xbmc/addons/addoninfo/AddonInfoBuilder.cpp

```diff
@@ -405,7 +407,19 @@ bool CAddonInfoBuilder::ParseXMLTypes(CAddonType& addonType, AddonInfoPtr info,
     if (library == nullptr)
       library = GetPlatformLibraryName(child);
     if (library != nullptr)
+    {
       addonType.m_libname = library;
+      // linux is different an has the version number after the suffix
+      static const std::regex libRegex("^.*" +
+                                       CCompileInfo::CCompileInfo::GetSharedLibrarySuffix() +
+                                       "\\.?[0-9]*\\.?[0-9]*\\.?[0-9]*$");
+      if (std::regex_match(library, libRegex))
+      {
+        info->SetBinary(true);
```

Comment:
does this really need to be public? Setting this kind of info should only be done when the addon is parsed so I don't see the need for exposing it

Reply:
How do you mean public?

Reply:
Removed public function as it's unused.